ansible_collections/arista/avd/plugins/plugin_utils/utils/range_expand.py

```diff
@@ -0,0 +1,147 @@
+
+    Example
+    -------
+    "Eth2-4" -> ["Eth2", "Eth3", "Eth4"]
+    "Eth(1,2,3)" -> ["Eth1", "Eth2", "Eth3"],
+    "Ethernet(4,1-2)" -> ["Ethernet4", "Ethernet1", "Ethernet2"]
+
+    Parameters
+    ----------
+    range_to_expand :
+        str | list to expand
+    prefix :
+        string, dafault value = ""
+
+    Returns
+    -------
+    list of expanded strings
+    """
+    if not (isinstance(range_to_expand, list) or isinstance(range_to_expand, str)):
+        raise AristaAvdError(f"value must be of type list or str, got {type(range_to_expand)}")
+
+    result = []
+
+    # If we got a list, unpack it and run this function recursively
+    if isinstance(range_to_expand, list):
+        for one_range in range_to_expand:
+            result.extend(range_expand_utils(one_range, prefix))
+
+    else:
+        # Must be a str now
+        if range_to_expand is None:
+            return
+
+        regex = r"([A-Za-z]+\s*)?(.*)"
+        sr = search(regex, range_to_expand)
+
+        prefix_group = sr.groups()
+        if prefix_group[0]:
+            prefix = prefix_group[0]
+        else:
+            range_to_expand = prefix + range_to_expand
+
+        regex_commas_outside_parentheses = r",(?![^(]*\))"
+        # Above regex matches all the commas "," outside parentheses.
+        # Except the case where there is no opening parentheses but closing parentheses is present [Example: ",4)"]
+        # It considers the comma inside the parentheses whereas it is actually outside.
+
+        regex_commas_inside_parentheses = r"(\(\s*(?:\d+)(?:(?:\s*,\s*\d+)*(?:,?(?:\d+)?\-\d+)*)*\s*\))"
+        #                           (\( \))                                                              matches starting and ending of parentheses
+        #                           ?:                                                                   only group but do not remember the grouped part
+        #                                         (?:\d+)                                                matches 1 or more numbers Ex. 1, 21
+        #                                                   (?:\s*,\s*\d+)*                              matches 0 or more apperance of "," and one
+        #                                                                                                             or more numbers Ex. ,2,23
+        #                                                                  (?:,?\d+\-\d+)*               matches 0 or more apperance of "," and one or more
+        #                                                                                                             numbers with range (-) Ex. ,2-4
+        #                                         (?:\d+)(?:(?:\s*,\s*\d+)*(?:,?\d+\-\d+)*)*             matches values similar to 1 or 1,2,3 or 1,2-4
+        #                                   (\(\s*(?:\d+)(?:(?:\s*,\s*\d+)*(?:,?\d+\-\d+)*)*\s*\))       matches values similar to 1 or 1,2,3 or 1,2-4
```

Comment:
This may be missing an update compared to line 121

Reply:
Updated